modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesContextInformationManager.java

```diff
@@ -435,19 +435,26 @@ public void removeBinding(InboundBindingEvent removeBindingEvent) throws AndesEx
         AndesMessageRouter messageRouter = AndesContext.getInstance().
                 getMessageRouterRegistry().getMessageRouter(messageRouterName);
 
-        messageRouter.removeMapping(removeBindingEvent.getBindingKey(), queue);
+        /*
+         * queue can be null if it is already removed. For non-durable topic we keep
+         * a single queue for all messages but Qpid issues binding remove calls for
+         * every internal queues it create for each subscriber
+         */
+        if ((null != queue) && (null != messageRouter)) {
+            messageRouter.removeMapping(removeBindingEvent.getBindingKey(), queue);
 
-        AndesBinding removedBinding = amqpConstructStore.removeBinding(messageRouterName, boundQueueName, true);
+            AndesBinding removedBinding = amqpConstructStore.removeBinding(messageRouterName, boundQueueName, true);
 
-        clusterNotificationAgent.notifyBindingsChange(removedBinding,
-                ClusterNotificationListener.BindingChange.Deleted);
+            clusterNotificationAgent.notifyBindingsChange(removedBinding,
+                    ClusterNotificationListener.BindingChange.Deleted);
 
-        //if a non durable queue on binding removal delete the queue if there are no more
-        // subscribers. Delete call for non durable queues is prevented at Qpid-Andes bridge.
-        if(!queue.isDurable() && queue.getBoundedSubscriptions().isEmpty()) {
-            InboundQueueEvent queueDeleteEvent = new InboundQueueEvent(queue.getName(),
-                    queue.isDurable(), queue.isShared(), queue.getQueueOwner(), queue.isExclusive());
-            deleteQueue(queueDeleteEvent);
+            //if a non durable queue on binding removal delete the queue if there are no more
+            // subscribers. Delete call for non durable queues is prevented at Qpid-Andes bridge.
+            if (!queue.isDurable() && queue.getBoundedSubscriptions().isEmpty()) {
+                InboundQueueEvent queueDeleteEvent = new InboundQueueEvent(queue.getName(),
+                        queue.isDurable(), queue.isShared(), queue.getQueueOwner(), queue.isExclusive());
+                deleteQueue(queueDeleteEvent);
+            }
         }
```

Comment:
This is by design. So if we keep a log it will be printed all over log file. It will not add any value.